### PR TITLE
[TS SDK] Add getBlock

### PR DESCRIPTION
**Pull request by zihan-aptos**
### Description
Add two functions for AptosClient sdk:
* getBlockByHeight
* getBlockByVersion
### Test Plan
Run a local node (run from the root of the repo):

```
cargo run -p aptos -- node run-local-testnet --with-faucet --faucet-port 8081 --force-restart --assume-yes
```

Run the SDK tests and make sure they pass. Go to the SDK directory, and setup an env to configure the URLs:

```
rm .env
echo 'APTOS_NODE_URL="http://127.0.0.1:8080/v1"' >> .env
echo 'APTOS_FAUCET_URL="http://127.0.01:8081"' >> .env
```

Run the tests:

```
yarn test
```

Screen Shot 2022-09-19 at 1 07 52 PM

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4287)

### CODE REVIEW

**Comment by zihan-aptos** [comment]
First time working on aptos-core. Any comments would be appreciated.

File: ecosystem/typescript/sdk/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ All notable changes to the Aptos Node SDK will be captured in this file. This ch
 ## Unreleased
 N/A
 
+## 1.3.14 (2022-09-16)
+
+- Added new functions `getBlockByHeight` and `getBlockByVersion`.
```
Comment (jjleng) [inline ecosystem/typescript/sdk/CHANGELOG.md:12]: Move this to `Unreleased`

File: ecosystem/typescript/sdk/package.json [hunk 2]
```diff
@@ -71,5 +71,5 @@
     "tsup": "6.2.3",
     "typescript": "4.8.2"
   },
-  "version": "1.3.13"
+  "version": "1.3.14"
```
Comment (jjleng) [inline ecosystem/typescript/sdk/package.json:74]: We are not releasing 1.3.14 yet. So no need to change this.

File: ecosystem/typescript/sdk/src/aptos_client.test.ts [hunk 3]
```diff
@@ -477,3 +477,18 @@ test(
...
+  expect(block.block_height).toBe(blockHeight.toString());
+});
+
+test("gets block by version", async () => {
+  const version = 100;
+  const client = new AptosClient(NODE_URL);
+  const block = await client.getBlockByVersion(version);
+  expect(parseInt(block.first_version)).toBeLessThanOrEqual(version);
+  expect(parseInt(block.last_version)).toBeGreaterThanOrEqual(version);
+});
```
Comment (jjleng) [inline ecosystem/typescript/sdk/src/aptos_client.test.ts:494]: Add a trailing empty line

File: ecosystem/typescript/sdk/src/aptos_client.ts [hunk 4]
```diff
@@ -767,6 +767,26 @@ export class AptosClient {
...
     return new HexString(origAddress);
   }
+
+  /**
+   * Get block by height
+   * @param blockHeight
+   * @param withTransactions
+   * @returns
+   */
+  async getBlockByHeight(blockHeight: number, withTransactions?:boolean): Promise<Gen.Block> {
```
[the remark this answers is not in the document]
Comment (zihan-aptos) [reply ecosystem/typescript/sdk/src/aptos_client.ts:777]: what's @parseApiError for?

File: ecosystem/typescript/sdk/src/aptos_client.test.ts [hunk 5]
```diff
@@ -477,3 +477,26 @@ test(
   },
   30 * 1000,
 );
+
+test(
+  "gets block by height", 
+  async () => {
+    const blockHeight = 100;
+    const client = new AptosClient(NODE_URL);
```
Comment (zihan-aptos) [inline ecosystem/typescript/sdk/src/aptos_client.test.ts:485]: @jjleng btw the test doesn't pass with `NODE_URL`. So as other tests in this file. I got the test passed with `"https://fullnode.devnet.aptoslabs.com/"`.

**Comment by banool** [comment]
Could you fill in your test plan here? If you check out the README there is a testing section that shows you how to test this with a local testnet.

**Comment by zihan-aptos** [comment]
> Could you fill in your test plan here? If you check out the README there is a testing section that shows you how to test this with a local testnet.

thanks. i was looking for test instruction.

**Comment by banool** [comment]
@jjleng This looks good from the perspective of the regular generated client, but shouldn't we be aiming to use BCS instead?

@zihan-aptos in either case, I think you'll need to make sure the following look good:
```
yarn fmt
yarn lint
```

**Comment by zihan-aptos** [comment]
```
yarn fmt
yarn lint
```

**Review by movekevin — APPROVED** [review]

**Review by msavir — APPROVED** [review]

**Review by jjleng — APPROVED** [review]

File: ecosystem/typescript/sdk/src/aptos_client.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (jjleng) [reply ecosystem/typescript/sdk/src/aptos_client.ts]: @parseApiError rethrows developer friendly errors.

**Pull request merged**